### DISCUSSION TITLE
Filtering donation forms, reports and donor lists based on donation form author/owner

### DISCUSSION THREAD

**Discussion by aneesascf** [impress-org/givewp#7423] [Developer Q&A] [answered]
Hi, 

My team partner with other charities to support them with fundraising efforts. We'd like to allow our partners to use givewp as we do; create forms, look at reports, see donor lists etc. However, we want to ensure each partner only sees the forms and data for forms they have created in givewp. I noticed there is no author/owner ID related to each form, but is there a clean way to limit access of forms/reports/functionality based on that type of rule? i.e. I should only see the forms, donations and donors related to forms I have created, not everyone else's. 

I was looking into using custom post meta to save the user id of each form creator - although I am not sure whether each givewp form is a type of post. It would be better if there was a form creator/author ID to be able filter on to limit access/visibility. 

Any help or guidance here would be greatly appreciated.

**Comment by kjohnson** [comment, marked as answer]
Hey @aneesascf.

Each Give Donation Form is a custom post type (`give_forms`).

Adding custom meta for an owner ID is one approach. You could then possibly filter the `get_posts` results based on that meta.

You might also consider creating a custom admin menu and that lists only that user's donation forms.
